backend/main.py

```
@@ -71,151 +137,348 @@ def get_spots(
+        user.icon_url = icon_url
+        db.commit()
+        
+        return {
+            "success": True,
+            "icon_url": icon_url,
+            "message": "User icon updated successfully"
+        }
+        
+    except HTTPException:
+        raise
+    except Exception:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)